tests/test_view_data_cli.py

```diff
@@ -111,5 +114,127 @@ def test_postcmd(self):
+
+    def tearDown(self):
+        try:
+            self.postgres.stop()
+        except AttributeError:
+            return
+
+    @patch("pepys_admin.view_data_cli.iterfzf", return_value="Datafiles")
+    def test_do_view_table(self, patched_input):
+        temp_output = StringIO()
+        with redirect_stdout(temp_output):
+            self.shell.do_view_table()
+        output = temp_output.getvalue()
+        print(output)
+        assert "Datafiles\n" in output
+        assert "| datafile_type_name   | reference                        | url   |\n" in output
+        assert "| E-Trac               | e_trac_bad.txt                   | None  |\n" in output
+        assert "| E-Trac               | e_trac.txt                       | None  |\n" in output
+        assert "| EAG                  | 20200305_ROBIN.eag.txt           | None  |" in output
+        assert "| NMEA                 | NMEA_bad.log                     | None  |" in output
+
+    @patch("pepys_admin.view_data_cli.iterfzf", return_value="alembic_version")
+    def test_do_view_table_alembic_version(self, patched_input):
+        temp_output = StringIO()
+        with redirect_stdout(temp_output):
+            self.shell.do_view_table()
+        output = temp_output.getvalue()
+        assert "Alembic Version\n" in output
+        assert "| version_number   |" in output
+
+    @patch("pepys_admin.view_data_cli.prompt", return_value='SELECT * FROM "pepys"."Datafiles";')
+    def test_do_run_sql(self, patched_prompt):
+        temp_output = StringIO()
+        with redirect_stdout(temp_output):
+            self.shell.do_run_sql()
+        output = temp_output.getvalue()
+        print(output)
+        assert 'SELECT * FROM "pepys"."Datafiles";' in output
+        assert (
+            "| e_trac_bad.txt                   | None | 5261 | 7bbe513d9d253d2277435e0849ed8342"
+            in output
+        )
+        assert (
+            "| e_trac.txt                       | None | 5315 | 577fad568cda2eb0b24178f5554f2b46"
+            in output
+        )
+        assert (
+            "| NMEA_bad.log                     | None |  243 | 8ddb840fee218872d2bb394cc654bdae"
+            in output
+        )
+
+    @patch("pepys_admin.view_data_cli.prompt", return_value="SELECT Blah from NonExisting")
+    def test_do_run_sql_invalid(self, patched_prompt):
+        temp_output = StringIO()
+        with redirect_stdout(temp_output):
+            self.shell.do_run_sql()
+        output = temp_output.getvalue()
+        print(output)
+        assert "ERROR: Query couldn't be executed successfully" in output
+
+    def test_do_cancel(self):
+        temp_output = StringIO()
+        with redirect_stdout(temp_output):
+            self.shell.do_cancel()
+        output = temp_output.getvalue()
+        assert "Returning to the previous menu..." in output
+
+    def test_default(self):
+        # Only cancel command (0) returns True, others return None
```

Comment:
I think there is a typo, it should be `command (.)`, right?

Reply:
Thanks, fixed now.